aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1048 @@
+    }
+
+    struct RemoveCommitteeEvent has store, drop {
+        committee_id: u64,
+        committee_info: CommitteeInfo
+    }
+
+    struct UpdateCommitteeEvent has store, drop {
+        committee_id: u64,
+        old_committee_info: CommitteeInfo,
+        new_committee_info: CommitteeInfo
+    }
+
+    struct AddCommitteeMemberEvent has store, drop {
+        committee_id: u64,
+        committee_member: DoraNodeInfo
+    }
+
+    struct RemoveCommitteeMemberEvent has store, drop {
+        committee_id: u64,
+        committee_member: DoraNodeInfo
+    }
+
+    struct UpdateNodeInfoEvent has store, drop {
+        committee_id: u64,
+        old_node_info: DoraNodeInfo,
+        new_node_info: DoraNodeInfo
+    }
+
+    struct CreateCommitteeInfoStoreEvent has store, drop {
+        committee_id: u64,
+        committee_info: CommitteeInfo
+    }
+    /// Internal - check if the node exists in the committee
+    fun does_node_exist(committee: &CommitteeInfo, node_address: address): bool {
+        simple_map::contains_key(&committee.map, &node_address)
+    }
+    /// Internal - Assert if the node exists in the committee
+    fun ensure_node_address_exist(committee: &CommitteeInfo, node_address: address) {
+        assert!(does_node_exist(committee, node_address), error::invalid_argument(NODE_NOT_FOUND))
+    }
+
+    /// Internal - create event handler
+    fun create_event_handler(owner_signer: &signer) {
+        move_to(owner_signer, SupraCommitteeEventHandler {
+            create: new_event_handle<CreateCommitteeInfoStoreEvent>(owner_signer),
+            add_committee: new_event_handle<AddCommitteeEvent>(owner_signer),
+            remove_committee: new_event_handle<RemoveCommitteeEvent>(owner_signer),
+            update_committee: new_event_handle<UpdateCommitteeEvent>(owner_signer),
+            add_committee_member: new_event_handle<AddCommitteeMemberEvent>(owner_signer),
+            remove_committee_member: new_event_handle<RemoveCommitteeMemberEvent>(owner_signer),
+            update_node_info: new_event_handle<UpdateNodeInfoEvent>(owner_signer),
+        });
+    }
+
+    /// Internal - create OwnerCap
+    fun create_owner(owner_signer: &signer) {
+        capability::create<OwnerCap>(owner_signer, &OwnerCap {});
+    }
+
+    /// Internal - create committeeInfo store functions
+    fun create_committeeInfo_store(owner_signer: &signer): signer {
+        let (resource_signer, _) = account::create_resource_account(owner_signer, SEED_COMMITTEE);
+        move_to(&resource_signer, CommitteeInfoStore {
+            committee_map: simple_map::new(),
+            node_to_committee_map: simple_map::new()
+        });
+        resource_signer
+    }
+
+    /// Its Initial function which will be executed automatically while deployed packages
+    fun init_module(owner_signer: &signer) {
+        create_owner(owner_signer);
+        create_committeeInfo_store(owner_signer);
+        create_event_handler(owner_signer);
```

Comment:
I believe we want to store all event handlers as well in the `resource_account` that got created in `create_committeeInfo_store`. Because of this, you may want to define `get_committeeInfo_address` and `get_committeeInfo_signer` since for accessing events you may need the address of this resource account frequently.

Reply:
Updated